From e65ac14d30368972c7cc40010765982263d8fcec Mon Sep 17 00:00:00 2001
From: Jani Hautakangas <jani@igalia.com>
Date: Thu, 12 Sep 2024 00:15:33 -0700
Subject: [PATCH 08/12] [WebDriver][WPE] Mouse interaction simulation needs to
 take accout device scale with old API
 https://bugs.webkit.org/show_bug.cgi?id=279438

Reviewed by Carlos Garcia Campos.

WebAutomationSession misses scaling mouse interaction simulation in case of old WPE API.

* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp:
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):

Canonical link: https://commits.webkit.org/283541@main
---
 .../UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp  | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp b/Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp
index 47da2f79a4e9..9b705fae9426 100644
--- a/Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp
+++ b/Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp
@@ -175,13 +175,14 @@ void WebAutomationSession::platformSimulateMouseInteraction(WebPageProxy& page,
 {
     UNUSED_PARAM(pointerType);
 
+    auto location = deviceScaleLocationInView(page, locationInView);
+
     if (page.viewBackend()) {
-        platformSimulateMouseInteractionLibWPE(page, interaction, button, locationInView, keyModifiers, pointerType, m_currentModifiers);
+        platformSimulateMouseInteractionLibWPE(page, interaction, button, location, keyModifiers, pointerType, m_currentModifiers);
         return;
     }
 
 #if ENABLE(WPE_PLATFORM)
-    auto location = deviceScaleLocationInView(page, locationInView);
     unsigned wpeButton = libWPEMouseButtonToWPEButton(button);
     auto modifier = libWPEStateModifierForWPEButton(wpeButton);
     uint32_t state = modifiersToEventState(keyModifiers) | m_currentModifiers;
-- 
2.43.0

